index.js

```diff
@@ -1,52 +1,72 @@
 const fs = require('fs');
 const path = require('path');
-const webpack = require('webpack');
+
+const entryPointNames = ['forum', 'admin'];
+const entryPointExts = ['js', 'ts'];
+
+function getEntryPoints() {
+  const entries = {};
+
+  appLoop: for (const app of entryPointNames) {
+    for (const ext of entryPointExts) {
+      const file = path.resolve(process.cwd(), `${app}.${ext}`);
+
+      if (fs.existsSync(file)) {
+        entries[app] = file;
+        continue appLoop;
+      }
+    }
+  }
+
+  if (Object.keys(entries).length === 0) {
+    console.error('ERROR: No JS entrypoints could be found.');
+  }
+
+  return entries;
+}
 
 module.exports = function (options = {}) {
   return {
     // Set up entry points for each of the forum + admin apps, but only
     // if they exist.
-    entry: (function () {
-      const entries = {};
+    entry: getEntryPoints(),
 
-      for (const app of ['forum', 'admin']) {
-        const file = path.resolve(process.cwd(), app + '.js');
-        if (fs.existsSync(file)) {
-          entries[app] = file;
-        }
-      }
-
-      return entries;
-    })(),
+    resolve: {
+      extensions: ['.ts', '.tsx', '.js', '.jsx', '.json'],
```

Comment:
Why json?

Reply:
Webpack/Babel can cleanly handle the importing of JSON.

This could be useful for an extension that wanted to load a large dataset and modify the frontend using that data.

Maybe some data exported from another system which is then used to add a page to the forum, like a leaderboard for a game server?